kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoScalaObjectMapper.scala

```diff
@@ -19,14 +19,54 @@ package org.apache.kyuubi.server.trino.api
 
 import javax.ws.rs.ext.ContextResolver
 
-import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.databind.{DeserializationFeature, JsonSerializer, Module, ObjectMapper, SerializerProvider}
+import com.fasterxml.jackson.databind.module.SimpleModule
 import com.fasterxml.jackson.datatype.jdk8.Jdk8Module
+import io.trino.client.{ClientTypeSignature, ClientTypeSignatureParameter}
 
 class TrinoScalaObjectMapper extends ContextResolver[ObjectMapper] {
 
   private lazy val mapper = new ObjectMapper()
     .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+    .registerModule(serializeTrinoClientTypeModule())
     .registerModule(new Jdk8Module)
 
+  /**
+   * Not sure why jackson annotations do not work for some Trino dto, e.g. `ClientTypeSignature`.
+   * Without this change, it will throw exception during serializing when return back response.
+   * This is a workaround to be compatible with Trino json serialization.
+   * TODO: Remove this when find root cause
+   */
+  private def serializeTrinoClientTypeModule(): Module = {
+    val module = new SimpleModule("trinoSerializer")
```

Comment:
![image](https://user-images.githubusercontent.com/5451385/219009080-7165ae14-f16e-4af3-9e02-6116dc316c61.png)
Can we refer the implementaion of `QueryResults` codec in trino client ?

Reply:
ah, good point. I have validated it works